### ISSUE TITLE
Turn off cache logging when --log-cache-events is not set

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#1139]
### Issue description
dbt has a `--log-cache-events` flag, but it appears that some invocations of dbt log the cache events even when this is flag not provided. Cache events should only be logged when `--log-cache-events` is provided.

### Results
`--log-cache-events` was not provided, but cache events were logged to `logs/dbt.log`

This happened for both `dbt run` and `dbt archive`, though it may apply to other types of invocations as well.

### System information
The output of `dbt --version`:
```
0.12.1
```

The operating system you're running on:
Amazon Linux

The python version you're using (probably the output of `python --version`)
3.5

### Steps to reproduce
TBD. In the logs I have which demonstrates this, I see:

```
running dbt with arguments Namespace(cls=, debug=False, log_cache_events=False, profile='user', profiles_dir='/tmp/jobs/508650/.dbt', strict=False, target='prod', threads=None, use_cache=True, vars='{}', which='archive')
```

followed by parsing, and then:

```
On master: select
table_name as name, table_schema as schema, table_type as type
from information_schema.tables
where table_schema ilike 'analytics'
SQL status: SUCCESS 70 in 1.15 seconds
Adding relation: _CachedRelation(schema=ANALYTICS, identifier=XXX, inner="ANALYTICS"."XXX")
before adding: {'ANALYTICS.XXX': [],
'ANALYTICS.YYY': [],
'ANALYTICS.ZZZ': [],
```

**Comment by beckjake** [comment]
This is another issue around calling `handle_and_check` twice (see #1138) / bypassing dbt's logging machinery. If you never call `dbt.logging.initialize_logger`, the cache logger will propagate by default due to how python works.

The safest solution here is to flip that flag during `handle_and_check` based on the given arguments, instead of during one-time logger initialization.

**Comment by beckjake** [comment]
Fixed in #1144
